### ISSUE TITLE
aws_route: IPv6 Route: Error finding route after creating it

### ISSUE THREAD

**Issue by benrobinsonsonos** [terraform-providers/terraform-provider-aws#3422]
### Terraform Version
terraform 0.11.3
terraform-provider-aws_v1.9.0_x4

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_route

### Terraform Configuration Files
```hcl

resource "aws_vpc" "vpc" {
  cidr_block = "10.0.0.0/16"
  assign_generated_ipv6_cidr_block = "true"
}

resource "aws_internet_gateway" "igw" {
  vpc_id = "${aws_vpc.vpc.id}"
}

resource "aws_subnet" "pub_sub_a" {
  vpc_id = "${aws_vpc.vpc.id}"
  cidr_block = "10.0.0.0/24"
  availability_zone = "us-east-1a"
}

resource "aws_route_table" "pub_rt" {
  vpc_id = "${aws_vpc.vpc.id}"
}

resource "aws_route_table_association" "pub_rta_a" {
  subnet_id = "${aws_subnet.pub_sub_a.id}"
  route_table_id = "${aws_route_table.pub_rt.id}"
}

resource "aws_route" "pub_rt_def" {
  route_table_id = "${aws_route_table.pub_rt.id}"
  destination_cidr_block = "0.0.0.0/0"
  gateway_id = "${aws_internet_gateway.igw.id}"
  depends_on = ["aws_internet_gateway.igw"]
}

resource "aws_route" "pub_rt_ipv6_def" {
  route_table_id = "${aws_route_table.pub_rt.id}"
  destination_ipv6_cidr_block = "::0/0"
  gateway_id = "${aws_internet_gateway.igw.id}"
  depends_on = ["aws_internet_gateway.igw"]
}

```

### Expected Behavior
IPv6 route should be created and added to the state.

### Actual Behavior
Terraform creates IPv6 route that is visible in management console, however, repeats, "Still creating..." until (2m0s elapsed), then returns an error that route couldn't be found after creating it. Repeating 'terraform apply' returns the same results.

* aws_route.pub_rt_ipv6_def: Error finding route after creating it: Unable to find matching route for Route Table (rtb-123456) and destination IPv6 CIDR block (::0/0).

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by smihajlovic-frame** [comment]
@benrobinsonsonos do you have any updates on this? I have similar issues with ipv4 routes

**Comment by benrobinsonsonos** [comment]
Same behavior observed in terraform-provider-aws_v1.10.0_x4. IPv4 route successfully created and added to the statefile. IPv6 route created, however, output reports "Error finding route after creating it."

**Comment by DrFaust92** [comment]
I was not able to reproduce this, is this still an issue?

**Comment by mmerickel** [comment]
I ran into this with a very similar setup. I think the issue is that AWS is canonicalizing the IPv6 route and terraform is not, so terraform is looking for `::0/0` and AWS made a route for `::/0` and terraform cannot find it after it was created.

**Comment by DrFaust92** [comment]
Can you share you're config so I'll try to reproduce?

**Comment by DrFaust92** [comment]
@mmerickel i've dived into the code a bit and it seems you are right

**Comment by DrFaust92** [comment]
opened #12890 to address this

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.59.0 of the Terraform AWS Provider, likely tomorrow. Thanks to @DrFaust92 for the implementation. 👍
